File: Level1/3진법 뒤집기.py
#3진법 뒤집기
def solution(n):
    answer =''
    result=0
    while n>=1: #반복문으로 3진수 구하기
        r=n%3
        n//=3
        answer+=str(r)
    i=0 #3의 승을 위해
    for idx in range(len(answer)-1,-1,-1): #리스트 탐색을 오른쪽->왼쪽으로 하기위해
        result+=int(answer[idx])*(3**i)  #각자리수*3^i
        i+=1
    return result
print(solution(45))

File: Level1/3진법 뒤집기.cpp
#include <string>
#include <vector>

using namespace std;

//3진법 변환
string get_ter(int n) {
    string tmp = "";
    
    while (n > 0) {
        int mod = n % 3;
        tmp += (char)(mod + '0');
        n /= 3;
    }
    //이 연산으로 뒤집어서 나옴
    return tmp;
}

//10진법 변환
int get_dec(string ter) {
    int tot = 0;
    int size = ter.size();
    
    for (int i = size - 1, j = 1; i>=0; --i, j *= 3)
        tot += j * (ter[i] - '0');
    
    return tot;
}

int solution(int n) {
    int answer = 0;
    answer = get_dec(get_ter(n));
    return answer;
}
